staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -284,6 +295,12 @@ func waitForDelete(pods []corev1.Pod, interval, timeout time.Duration, usingEvic
 		}
 		pods = pendingPods
 		if len(pendingPods) > 0 {
+			select {
+			case <-ctx.Done():
+				return false, fmt.Errorf("global timeout")
```

Comment:
would be useful to include the time if it is easy to get at.

Reply:
We don't log the time elsewhere, and it's a time the user sets.

Reply:
1. not doing it elsewhere doesn't mean it isn't valuable.  the intent of this pr is to enable this to be used in new contexts where this is more important.
2. the user that sets it may not be the user needing to debug it if something goes wrong.  if this is being used as a library and called from a server, ideally a user debugging the system that sees the error shouldn't have to look elsewhere for the timeout.

Reasons not to do this would be -- its not readily available and would cause a lot of disruptive changes to the code.

Reply:
@pwittrock Okay, I added the timeout info.  IMO, it's a little ugly to pass that to waitfordelete.

Reply:
understood.  I think it is worth the tradeoff.  informative error reporting can be a lifesaver.